backend/fbdev.c

```diff
@@ -220,9 +294,20 @@ twin_context_t *twin_fbdev_init(int width, int height)
         goto bail_vt_fd;
     }
 
+    /* Examine if framebuffer mapping is valid */
+    if (tx->fb_base == MAP_FAILED) {
+        log_error("Failed to map framebuffer memory");
+        return;
+    }
+
     /* Create TWIN screen */
-    ctx->screen =
-        twin_screen_create(width, height, NULL, _twin_fbdev_put_span, ctx);
+    ctx->screen = twin_screen_create(
+        width, height, NULL,
+        (tx->fb_var.bits_per_pixel == 16)   ? _twin_fbdev_put_span16
+        : (tx->fb_var.bits_per_pixel == 24) ? _twin_fbdev_put_span24
+        : (tx->fb_var.bits_per_pixel == 32) ? _twin_fbdev_put_span32
+                                            : NULL,
+        ctx);
```

Comment:
Consider the changes below:
```diff
@@ -300,13 +300,14 @@ twin_context_t *twin_fbdev_init(int width, int height)
         return;
     }
 
+    const twin_put_span_t fbdev_put_spans[] = {
+        _twin_fbdev_put_span16,
+        _twin_fbdev_put_span24,
+        _twin_fbdev_put_span32,
+    };
     /* Create TWIN screen */
     ctx->screen = twin_screen_create(
-        width, height, NULL,
-        (tx->fb_var.bits_per_pixel == 16)   ? _twin_fbdev_put_span16
-        : (tx->fb_var.bits_per_pixel == 24) ? _twin_fbdev_put_span24
-        : (tx->fb_var.bits_per_pixel == 32) ? _twin_fbdev_put_span32
-                                            : NULL,
+        width, height, NULL, fbdev_put_spans[tx->fb_var.bits_per_pixel / 8 - 2],
         ctx);
 
     /* Create Linux input system object */
```

Reply:
I would like to know if it is necessary to check `tx->fb_var.bits_per_pixel` to prevent index out of bounds.

Reply:
> I would like to know if it is necessary to check `tx->fb_var.bits_per_pixel` to prevent index out of bounds.

You did check the supported bpp earlier.